project/KeywordPlugin.scala

```diff
@@ -0,0 +1,34 @@
+import sbt._
+import Keys._
+
+object KeywordPlugin extends AutoPlugin {
+  override val requires = plugins.JvmPlugin
+
+  lazy val scalaKeywords = TaskKey[Set[String]]("scala-keywords")
+  lazy val generateKeywords = TaskKey[File]("generateKeywords")
+
+  def getScalaKeywords: Set[String] =
+    {
+      val g = new scala.tools.nsc.Global(new scala.tools.nsc.Settings)
+      g.nme.keywords.map(_.toString)
```

Comment:
I think it won't happen during the entire life cycle of 2.x so we're good for a while.